module/project/templateTasks/helper.js

```diff
@@ -0,0 +1,638 @@
+            try {
+                
+                let queryObject = (filterData != "all") ? filterData : {};
+                let projection = {}
+           
+                if (fieldsArray != "all") {
+                    fieldsArray.forEach(field => {
+                        projection[field] = 1;
+                   });
+               }
+               
+               if( skipFields !== "none" ) {
+                   skipFields.forEach(field=>{
+                       projection[field] = 0;
+                   });
+               }
+               
+               let tasks = 
+               await database.models.projectTemplateTasks.find(
+                   queryObject, 
+                   projection
+               ).lean();
+           
+               return resolve(tasks);
+           
+           } catch (error) {
+               return reject(error);
+           }
+       });
+    }
+
+    /**
+     * Extract csv information.
+     * @method
+     * @name extractCsvInformation
+     * @param {Array} csvData - csv data.
+     * @param {String} projectTemplateId - project template id.
+     * @returns {Array} Lists of tasks.
+     */
+    
+    static extractCsvInformation(
+        csvData,
+        projectTemplateId
+    ) {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                let taskIds = [];
+                let solutionIds = [];
+                let systemId = false;
+
+                csvData.forEach(data=>{
+                    
+                    let parsedData = UTILS.valueParser(data);
+
+                    if( parsedData._SYSTEM_ID ) {
+                        taskIds.push(parsedData._SYSTEM_ID);
+                        systemId = true;
+                    } else {
+                        taskIds.push(parsedData.externalId);
+                    }
+
+                    if ( parsedData.solutionId && parsedData.solutionId !== "" ) {
+                        solutionIds.push(parsedData.solutionId);
+                    }
+                });
+
+                let tasks = {};
+
+                if( taskIds.length > 0 ) {
+
+                    let filterData = {};
+
+                    if( systemId ) {
+                        filterData = {
+                            _id : { $in : taskIds }
+                        }
+                    } else {
+                        filterData = {
+                            externalId : { $in : taskIds }
+                        }
+                    }
+
+                    let tasksData = await this.taskDocuments(
+                        filterData,
+                        ["_id","children","externalId","projectTemplateId","parentId"]
+                    );
+
+                    if( tasksData.length > 0 ) {
+                        tasksData.forEach(task=> {
+                            if( systemId ) {
+                                tasks[task._id.toString()] = task;
```

Comment:
@aman-tunerlabs - Not a good idea that key can be external or internal id , doesn't give a clear picture what to expect.